dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/programrule/engine/ProgramRuleEngineTest.java

```diff
@@ -458,6 +493,34 @@ void testAssignValueTypeAge()
         assertEquals( ruleEffects.get( 0 ).data(), "10" );
     }
 
+    @Test
+    void testAssignOptionSetName()
+    {
+        setUpRuleActionForOptionSetName();
+
+        Event programStageInstance = eventService.getEvent( "UID-PS13" );
+        List<RuleEffect> ruleEffects = programRuleEngine.evaluate( programStageInstance.getProgramInstance(),
+            programStageInstance, Sets.newHashSet(), List.of( programRuleToTestOptionSet ) );
+
+        assertNotNull( ruleEffects );
+        assertInstanceOf( RuleActionShowWarning.class, ruleEffects.get( 0 ).ruleAction() );
+        assertThat( ruleEffects.get( 0 ).data(), Matchers.is( option1.getName() ) );
+    }
+
+    @Test
+    void testAssignOptionSetCode()
+    {
+        setUpRuleActionForOptionSetCode();
+
+        Event programStageInstance = eventService.getEvent( "UID-PS13" );
+        List<RuleEffect> ruleEffects = programRuleEngine.evaluate( programStageInstance.getProgramInstance(),
+            programStageInstance, Sets.newHashSet(), List.of( programRuleToTestOptionSet ) );
+
+        assertNotNull( ruleEffects );
+        assertInstanceOf( RuleActionShowError.class, ruleEffects.get( 0 ).ruleAction() );
+        assertThat( ruleEffects.get( 0 ).data(), Matchers.is( option1.getCode() ) );
```

Comment:
`assertEquals` should probably do as described in https://github.com/dhis2/wow-backend/blob/master/guides/testing.md#assertions